modules/git/commit_reader.go

```diff
@@ -27,56 +27,70 @@ func CommitFromReader(gitRepo *Repository, sha plumbing.Hash, reader io.Reader)
 	pgpsig := false
 
 	scanner := bufio.NewScanner(reader)
+	// Split by '\n' but include the '\n'
+	scanner.Split(func(data []byte, atEOF bool) (advance int, token []byte, err error) {
+		if atEOF && len(data) == 0 {
+			return 0, nil, nil
+		}
+		if i := bytes.IndexByte(data, '\n'); i >= 0 {
+			// We have a full newline-terminated line.
+			return i + 1, data[0 : i+1], nil
+		}
+		// If we're at EOF, we have a final, non-terminated line. Return it.
+		if atEOF {
+			return len(data), data, nil
+		}
+		// Request more data.
+		return 0, nil, nil
```

Comment:
This is what I was saying about binary files. When you get to the paylod part, there could be several MB of data without hitting a `\n`. I don't see an immediate solution for it, so we've got to live with that. Also, large commits will also be held in memory anyway. 😬

Reply:
git actually is no better take a look at:

https://github.com/git/git/blob/228f53135a4a41a37b6be8e4d6e2b6153db4a8ed/commit.c#L1005-L1043

Reply:
OK, I didn't realize that in this case the payload consisted of only the commit header; I thought that the full content of the commit was being parsed. 👍